### ISSUE TITLE
Don't include log4j.properties in published JAR

### ISSUE THREAD

**Issue by ryan-williams** [bigdatagenomics/adam#1300]
adam-core (cf. [0.20.3](http://search.maven.org/remotecontent?filepath=org/hammerlab/adam/adam-core_2.11/0.20.3/adam-core_2.11-0.20.3.jar)) and utils-misc (cf. [0.2.10](http://search.maven.org/remotecontent?filepath=org/bdgenomics/utils/utils-misc_2.11/0.2.10/utils-misc_2.11-0.2.10.jar); and possibly others?) include (identical) `log4j.properties` in their JARs, leading to some unnecessary logic downstream when assembling uber-JARs that include both.

Specifically, I have to choose to drop one or the other when building an assembly jar in SBT:

```scala
assemblyMergeStrategy in assembly := {
  // Two org.bdgenomics deps include the same log4j.properties.
  case PathList("log4j.properties") => MergeStrategy.first
  case x => (assemblyMergeStrategy in assembly).value(x)
}
```

cf. [guacamole](https://github.com/hammerlab/guacamole/blob/9d330aeb3a7a040c174b851511f19b42d7717508/build.sbt#L38-L42), [filter-bam](https://github.com/hammerlab/filter-bam/blob/651f42e89a6e34e61af4bef1188698b634fa5f1b/build.sbt#L24-L28), [pageant](https://github.com/ryan-williams/pageant/blob/fc7f3b4066c231092cb3324422f27d2e5367e1a5/build.sbt#L26-L30).

I'm not sure what similar downstream Maven-configs do in the presence of multiple files like this, but it seems like logging configuration, in general, should not be included in packaged JARs? I've also recently seen chatter about issues around this in Hadoop-land, e.g. [HADOOP-7468](https://issues.apache.org/jira/browse/HADOOP-7468).

**Comment by fnothaft** [comment]
Yeah, agreed. Perhaps the best compromise between not spamming logs by default and not pushing properties files would be to put the log4j.properties in the:

* test resources
* adam-assembly jar

We push adam-assembly to Maven, but it isn't intended to be used for downstream development, so I think that'd be OK. Thoughts?

**Comment by ryan-williams** [comment]
Cool, this isn't something I've thought much about but I guess a rule like "executable/assembly JARs should come with relevant configs while library JARs should not" makes sense.

OTOH I could kind of see a world where even libraries shipped with some default logging config that only set default levels for their own classes/packages, and the contract was that downstream assemblers should have configuration that just appends all log4j.properties files.. I could put that into my parent project configuration in a non-BDG-dep-specific way and be done with it.

Your call / not urgent either way.

**Comment by fnothaft** [comment]
Nice idea! I think that's a good approach. Let's get 0.21.0 out first, and then we can get that change done.

**Comment by ryan-williams** [comment]
I just noticed that I think this is causing one of my downstream libraries' tests to write out `adam.log` files, even though it's not doing anything ADAM-related, due to [this line](https://github.com/bigdatagenomics/adam/blob/adam-parent_2.11-0.20.0/adam-core/src/main/resources/log4j.properties#L14).

Not a big deal, just noting; I'd never known where that file was coming from.

**Comment by fnothaft** [comment]
@heuermh is pretty sure this will be fixed by #1360.

**Comment by fnothaft** [comment]
Resolved by #1360.
